CoreUtilities/src/au/gov/asd/tac/constellation/utilities/log/ConstellationLogFormatter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2010-2024 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.utilities.log;
+
+import au.gov.asd.tac.constellation.utilities.text.SeparatorConstants;
+import java.text.MessageFormat;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.logging.Formatter;
+import java.util.logging.LogRecord;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ *
+ * @author Guilty-Spark-343
+ */
+public class ConstellationLogFormatter extends Formatter {
+    
+    private String logMessage = "";
+    private Integer logCount = 0;
+    
+    @Override
+    public final String format(final LogRecord logRecord) {
+        
+        final Throwable error = logRecord.getThrown();
+        
+        final StringBuilder sb = new StringBuilder();
+        sb.append("[");
+        sb.append(LocalDateTime.now().toString().substring(0,22).replace("T"," "));
+        sb.append("Z] - ");
+        sb.append(logRecord.getLevel());
+        sb.append(" [");
+        sb.append(logRecord.getLoggerName());
+        sb.append("]: ");
+        
+        final String formattedMessage;
+        
+        if (logRecord.getMessage() != null && logRecord.getParameters() != null) {
+            formattedMessage = MessageFormat.format(logRecord.getMessage(), logRecord.getParameters());
+        } else {
+            formattedMessage = logRecord.getMessage();
+        }
+
+        if(formattedMessage != null && logMessage != null && formattedMessage.equals(logMessage)){
+            if(logCount == 10){
+                sb.append("Last record repeated more than 10 times, further logs of this record are ignored until the log record changes.");
+            }else if(logCount > 10){
+                return ""; // Do not log messages until it changes
+            }else{
+                sb.append("Last record repeated again.");
+            }
+            logCount += 1;
+            return sb.toString();
+        }else {
+            logCount = 0;            
+        }
```

Comment:
Code formatting: add spaces between keywords and brackets (can all be fixed with Alt-Shift-F in the IDE)